tests/test.hs

```diff
@@ -146,6 +146,19 @@ prop_alt_help = once $
       i = info (p <**> helper) idm
   in checkHelpText "alt" i ["--help"]
 
+prop_optional_help :: Property
+prop_optional_help = once $
+  let p :: Parser (Maybe (String, String))
+      p = optional ((,)
+                    <$> strOption ( long "a"
+                                    <> metavar "A"
+                                    <> help "value a" )
+                    <*> strOption ( long "b"
+                                    <> metavar "B"
+                                    <> help "value b" ) )
+      i = info (p <**> helper) idm
+  in checkHelpText "optional" i ["--help"]
```

Comment:
Can you add the example from #307 showing nested optional inputs as well?

I've run it, it works, but it would be good to have it addressed in the tests too.

Reply:
Sorry, I mean #220.

Reply:
Done